internal/utils/main.go

```diff
@@ -562,3 +562,48 @@ func GetEnvironmentDetails(isTest bool) (details EnvironmentDetails) {
 		return details
 	}
 }
+
+type CaptiveCore interface {
+	CreateCaptiveCoreBackend() (ledgerbackend.CaptiveStellarCore, error)
+	GetLedgerCloseMeta(end uint32) (xdr.LedgerCloseMeta, error)
+}
+
+func (e EnvironmentDetails) CreateCaptiveCoreBackend() (*ledgerbackend.CaptiveStellarCore, error) {
+	captiveCoreToml, err := ledgerbackend.NewCaptiveCoreTomlFromFile(
+		e.CoreConfig,
+		ledgerbackend.CaptiveCoreTomlParams{
+			NetworkPassphrase:  e.NetworkPassphrase,
+			HistoryArchiveURLs: e.ArchiveURLs,
+			Strict:             true,
+		},
+	)
+	backend, err := ledgerbackend.NewCaptive(
+		ledgerbackend.CaptiveCoreConfig{
+			BinaryPath:         e.BinaryPath,
+			Toml:               captiveCoreToml,
+			NetworkPassphrase:  e.NetworkPassphrase,
+			HistoryArchiveURLs: e.ArchiveURLs,
+		},
+	)
+	return backend, err
+}
+
+func (e EnvironmentDetails) GetLedgerCloseMeta(end uint32) (xdr.LedgerCloseMeta, error) {
+	ctx := context.Background()
+
+	backend, err := e.CreateCaptiveCoreBackend()
+
+	ledgerCloseMeta, err := backend.GetLedger(ctx, end)
+	if err != nil {
+		return xdr.LedgerCloseMeta{}, err
+	}
+
+	return ledgerCloseMeta, nil
+}
+
+func GetCloseTimeV(ledger xdr.LedgerCloseMeta, v bool) (time.Time, error) {
+	if v == false {
+		return ExtractLedgerCloseTime(ledger.V0.LedgerHeader)
+	}
+	return ExtractLedgerCloseTime(ledger.V1.LedgerHeader)
+}
```

Comment:
I think we should wait until the latest xdr changes are merged into the master branch. There will be some new functions to help parse different `LedgerCloseMeta` types, like this `LedgerHeaderHistoryEntry()` [here](https://github.com/stellar/go/blob/soroban-xdr-next/xdr/ledger_close_meta.go#L5-L16)

```
func (l LedgerCloseMeta) LedgerHeaderHistoryEntry() LedgerHeaderHistoryEntry {
	switch l.V {
	case 0:
		return l.MustV0().LedgerHeader
	case 1:
		return l.MustV1().LedgerHeader
	case 2:
		return l.MustV2().LedgerHeader
	default:
		panic(fmt.Sprintf("Unsupported LedgerCloseMeta.V: %d", l.V))
	}
}

```


Then you'll just be able to pass a generic `LedgerHeader` and extract the ledger close time regardless of ledger close type.

Reply:
Update: we'll leave this as a to-do to refactor with the testnet release.